packages/hebao_v1/contracts/modules/core/WalletFactoryModule.sol

```diff
@@ -50,39 +50,34 @@ contract WalletFactoryModule is WalletFactory, MetaTxModule
         public
         MetaTxModule(_controller)
     {
-
     }
 
     /// @dev Create a new wallet by deploying a proxy.
-    /// @param _expectedWallet The expected smart contract wallet address that will be created.
     /// @param _owner The wallet's owner.
     /// @param _modules The wallet's modules.
     /// @param _subdomain The ENS subdomain to register, use "" to skip.
     /// @return _wallet The newly created wallet's address.
     function createWallet(
-        address            _expectedWallet,
```

Comment:
I added this here to make it compatible with meta transactions without doing anything as these currently always get the wallet address from the first function parameter. But this can also be done quite easily by overriding `extractWalletAddress` in `WalletFactoryModule`:

```
function extractWalletAddress(bytes memory data)
    internal
    view
    returns (address wallet)
{
    address owner = extractAddressFromCallData(data, 0);
    wallet = computeWalletAddress(owner);
}
```